internal/configs/version1/nginx-plus.ingress.tmpl

```diff
@@ -73,7 +73,28 @@ server {
 	{{range $AppProtectLogConf := $server.AppProtectLogConfs}}app_protect_security_log {{$AppProtectLogConf}};
 	{{end}}
 	{{- end}}
-	
+
+	{{- if $server.AppProtectDosEnable}}
+	app_protect_dos_enable {{$server.AppProtectDosEnable}};
+	{{if $server.AppProtectDosPolicy}}app_protect_dos_policy_file {{$server.AppProtectDosPolicy}};{{end}}
+	{{- end}}
+	{{- if $server.AppProtectDosLogEnable}}
+	app_protect_dos_security_log_enable {{$server.AppProtectDosLogEnable}};
+	{{if $server.AppProtectDosLogConf}}app_protect_dos_security_log {{$server.AppProtectDosLogConf}};{{end}}
+	    {{- if $server.AppProtectDosAccessLogDst}}
+        set $loggable '0';
+        access_log syslog:server={{ .AppProtectDosAccessLogDst }} log_dos if=$loggable;
+        {{- end}}
+	{{- end}}
+
+	{{- if $server.AppProtectDosMonitor}}
+	app_protect_dos_monitor "{{$server.AppProtectDosMonitor}}";
```

Comment:
could you possibly explain how this directive works?

for example, let's say we have `app_protect_dos_monitor example.com`

Does this mean the DOS component will resolve this DNS name and send requests to it? 
What resolver will be used?
If multiple IPs are resolved, what IP will be used?
What will happen if the resolved DNS record expires?

Is it correct to assume that the expectation that a monitor request will go through an Ingress Controller pod? 

I'm trying to understand if this directive can be a source of any failures. For example:
- What will happen if the DNS cannot be resolved (1) no DNS record (2) the DNS server is unavailable?
- What will happen if the resolved IP address (1) doesn't respond, (2) responds with an error?

Reply:
Lets take this example:
server {
   server_name coffee.com;
   listen 80;
   location /abc {
    app_protect_dos_monitor coffee.com/abc;
    proxy_pass be;
   }
}

Result: curl -H "Host: coffee.com" http://localhost/abc

So, the request go to localhost and get to the nginx process which send it to the proxy_pass.
There is no DNS resolved.
In case there is no respond or error respond it will be handle by the admd

Reply:
thanks for the clarification!

Although, I still see cases when it might not work. But it should work in most of the cases.

Most of Ingress or VirtualServer resources have TLS termination enabled with a HTTPs-redirect if a request comes to an HTTP port. So I think the user will need to monitor `https://`.

If you monitor `https://`, I don't think it will work in the following cases:
- TLS passthrough is enabled (https://docs.nginx.com/nginx-ingress-controller/configuration/transportserver-resource/ ), because in that case, NGINX config will require that client HTTPs connections include the [SNI](https://en.wikipedia.org/wiki/Server_Name_Indication), which the shown curl command doesn't set 
- when client authentication is enabled. For example, using https://docs.nginx.com/nginx-ingress-controller/configuration/policy-resource/#ingressmtls policy.

However, those cases are not as common.